public/app/features/alerting/components/AlertingQueryEditor.tsx

```diff
@@ -1,49 +1,70 @@
 import React, { PureComponent } from 'react';
 import { css } from '@emotion/css';
-import { DataSourceApi, GrafanaTheme } from '@grafana/data';
+import {
+  DataQuery,
+  getDefaultRelativeTimeRange,
+  GrafanaTheme2,
+  LoadingState,
+  PanelData,
+  RelativeTimeRange,
+} from '@grafana/data';
 import { selectors } from '@grafana/e2e-selectors';
 import { Button, HorizontalGroup, Icon, stylesFactory, Tooltip } from '@grafana/ui';
-import { config, getDataSourceSrv } from '@grafana/runtime';
+import { config } from '@grafana/runtime';
 import { AlertingQueryRows } from './AlertingQueryRows';
 import { dataSource as expressionDatasource, ExpressionDatasourceUID } from '../../expressions/ExpressionDatasource';
 import { getNextRefIdChar } from 'app/core/utils/query';
 import { defaultCondition } from '../../expressions/utils/expressionTypes';
 import { ExpressionQueryType } from '../../expressions/types';
 import { GrafanaQuery } from 'app/types/unified-alerting-dto';
+import { AlertingQueryRunner } from '../state/AlertingQueryRunner';
+import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
+import { isExpressionQuery } from 'app/features/expressions/guards';
 
 interface Props {
   value?: GrafanaQuery[];
   onChange: (queries: GrafanaQuery[]) => void;
 }
 
 interface State {
-  defaultDataSource?: DataSourceApi;
+  panelDataByRefId: Record<string, PanelData>;
 }
 export class AlertingQueryEditor extends PureComponent<Props, State> {
```

Comment:
I have also added a todo in the epic to cover this component with react testing library tests.